src/relations/ui.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Define the Temporal server ui relation."""
+
+import logging
+
+from ops import framework
+from ops.model import ActiveStatus
+
+from log import log_event_handler
+
+logger = logging.getLogger(__name__)
+
+
+class UI(framework.Object):
+    """Client for ui:temporal relations."""
+
+    def __init__(self, charm):
+        """Construct.
+
+        Args:
+            charm: The charm to attach the hooks to.
+        """
+        super().__init__(charm, "ui")
+        self.charm = charm
+        charm.framework.observe(charm.on.ui_relation_joined, self._on_ui_relation_joined)
+        charm.framework.observe(charm.on.ui_relation_changed, self._on_ui_relation_joined)
+
+    @log_event_handler(logger)
+    def _on_ui_relation_joined(self, event):
+        """Handle new ui:temporal relations.
+
+        Attempt to provide server status to the ui unit.
+
+        Args:
+            event: The event triggered when the relation changed.
+        """
+        if self.charm.unit.is_leader():
+            self._provide_server_status()
+
+    def _provide_server_status(self):
+        """Provide server status to the UI charm."""
+        charm = self.charm
+        server_status = charm.model.unit.status == ActiveStatus()
```

Comment:
`is_active`?

Reply:
Will merge for now and update in the next PR.